libs/api-client/src/api-client.ts

```diff
@@ -229,8 +242,16 @@ export class GFW_API_CLASS {
     return
   }
 
+  generateUrl(url: string, version?: ApiVersion): string {
+    if (isUrlAbsolute(url)) {
+      return url
+    }
```

Comment:
what about checking if the url already starts with a ApiVersion value with something like:
```
if (url.startsWith('v1') || url.startsWith('v2')) {
  return url
}
```
so we avoid [cleaning it by hand](https://github.com/GlobalFishingWatch/frontend/pull/1546/files#diff-c21edc9c2057bed8019b6df684a7685168b615f521d45c9f6ffc1bc8e62f5a41R131) in the case where the urls are autogenerated by the datasets endpoints definition

Reply:
It makes sense to me, I'd check it against the version prefix that would be added
```typescript
    const apiVersion = version ?? this.apiVersion
    const prefix = apiVersion ? `/${apiVersion}` : ''
    if (url.startsWith(`${prefix}/`)) {
      return url
    }
    return `${prefix}${url}`
```